android/capacitor/src/main/java/com/getcapacitor/CapacitorWebView.java

```diff
@@ -54,19 +55,35 @@ public boolean dispatchKeyEvent(KeyEvent event) {
         return super.dispatchKeyEvent(event);
     }
 
-    private void edgeToEdgeHandler() {
-        ViewCompat.setOnApplyWindowInsetsListener(this, (v, windowInsets) -> {
-            Insets insets = windowInsets.getInsets(WindowInsetsCompat.Type.systemBars());
+    public void edgeToEdgeHandler(Bridge bridge) {
+        String configEdgeToEdge = bridge.getConfig().adjustMarginsForEdgeToEdge();
 
-            MarginLayoutParams mlp = (MarginLayoutParams) v.getLayoutParams();
-            mlp.leftMargin = insets.left;
-            mlp.bottomMargin = insets.bottom;
-            mlp.rightMargin = insets.right;
-            mlp.topMargin = insets.top;
-            v.setLayoutParams(mlp);
+        if (configEdgeToEdge.equals("disable")) return;
 
-            // Don't pass window insets to children
-            return WindowInsetsCompat.CONSUMED;
-        });
+        boolean autoMargins = false;
+        boolean forceMargins = configEdgeToEdge.equals("force");
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
+            TypedValue value = new TypedValue();
+            boolean foundOptOut = getContext().getTheme().resolveAttribute(android.R.attr.windowOptOutEdgeToEdgeEnforcement, value, true);
+            boolean optOutValue = value.data != 0; // value is set to -1 on true as of Android 15, so we have to do this.
+
+            autoMargins = configEdgeToEdge.equals("auto") && !(foundOptOut && optOutValue);
```

Comment:
the `configEdgeToEdge.equals("auto")` could be moved to the if to avoid executing the code if set to force